onnxruntime/core/providers/nuphar/scripts/symbolic_shape_infer.py

```diff
@@ -1038,6 +1041,38 @@ def _infer_Slice(self, node):
             assert len(ends) == 1
             self.sympy_data_[node.output[0]] = self.sympy_data_[node.input[0]][starts[0]:ends[0]]
 
+    def _infer_Gelu(self, node):
+        data_shape = self._get_shape(node, 0)
+        vi = self.known_vi_[node.output[0]]
+        vi.CopyFrom(helper.make_tensor_value_info(node.output[0], onnx.TensorProto.FLOAT, data_shape))
+        self.known_vi_[vi.name] = vi
+
+    def _infer_LayerNormalization(self, node):
+        data_shape = self._get_shape(node, 0)
+        vi = self.known_vi_[node.output[0]]
+        vi.CopyFrom(helper.make_tensor_value_info(node.output[0], onnx.TensorProto.FLOAT, data_shape))
+        self.known_vi_[vi.name] = vi
+
+        data_shape[len(data_shape) - 1] = 1
+        vi = self.known_vi_[node.output[1]]
+        vi.CopyFrom(helper.make_tensor_value_info(node.output[1], onnx.TensorProto.FLOAT, data_shape))
+        self.known_vi_[vi.name] = vi
+
+        vi = self.known_vi_[node.output[2]]
+        vi.CopyFrom(helper.make_tensor_value_info(node.output[2], onnx.TensorProto.FLOAT, data_shape))
+        self.known_vi_[vi.name] = vi
+
+    def _infer_SoftmaxCrossEntropyLoss(self, node):
+        vi = self.known_vi_[node.output[0]]
+        vi.type.tensor_type.elem_type = onnx.TensorProto.FLOAT
+        self.known_vi_[vi.name] = vi
+
+        if len(node.output) > 1:
+            data_shape = self._get_shape(node, 0)
+            vi = self.known_vi_[node.output[1]]
+            vi.CopyFrom(helper.make_tensor_value_info(vi.name, onnx.TensorProto.FLOAT, data_shape))
+            self.known_vi_[vi.name] = vi
```

Comment:
May add input/output index to self._pass_on_shape_and_type() and reuse that code.